tools/protovalidate-conformance/internal/cases/cases_strings.go

```diff
@@ -1402,6 +1402,130 @@ func stringSuite() suites.Suite {
+					Rule:   results.FieldPath("string.ulid"),
+					RuleId: proto.String("string.ulid_empty"),
+				},
+			),
+		},
+		"ulid/invalid/not_checked/empty": {
+			Message:  &cases.StringNotULID{Val: ""},
+			Expected: results.Success(true),
+		},
+		"ulid/invalid/not_checked/malformed": {
+			Message:  &cases.StringNotULID{Val: "foobar"},
+			Expected: results.Success(true),
+		},
+		"ulid/invalid/malformed": {
+			Message: &cases.StringULID{Val: "foobar"},
+			Expected: results.Violations(
+				&validate.Violation{
+					Field:  results.FieldPath("val"),
+					Rule:   results.FieldPath("string.ulid"),
+					RuleId: proto.String("string.ulid"),
+				},
+			),
+		},
+		"ulid/invalid/too_short": {
+			Message: &cases.StringULID{Val: "01ARZ3NDEKTSV4RRFFQ69G5FA"},
+			Expected: results.Violations(
+				&validate.Violation{
+					Field:  results.FieldPath("val"),
+					Rule:   results.FieldPath("string.ulid"),
+					RuleId: proto.String("string.ulid"),
+				},
+			),
+		},
+		"ulid/invalid/too_long": {
+			Message: &cases.StringULID{Val: "01ARZ3NDEKTSV4RRFFQ69G5FAVX"},
+			Expected: results.Violations(
+				&validate.Violation{
+					Field:  results.FieldPath("val"),
+					Rule:   results.FieldPath("string.ulid"),
+					RuleId: proto.String("string.ulid"),
+				},
+			),
+		},
+		"ulid/invalid/invalid_characters/i": {
+			Message: &cases.StringULID{Val: "01ARZ3NDEKTSV4RRFFQ69G5FAI"},
+			Expected: results.Violations(
+				&validate.Violation{
+					Field:  results.FieldPath("val"),
+					Rule:   results.FieldPath("string.ulid"),
+					RuleId: proto.String("string.ulid"),
+				},
+			),
+		},
+		"ulid/invalid/invalid_characters/l": {
+			Message: &cases.StringULID{Val: "01ARZ3NDEKTSV4RRFFQ69G5FAL"},
+			Expected: results.Violations(
+				&validate.Violation{
+					Field:  results.FieldPath("val"),
+					Rule:   results.FieldPath("string.ulid"),
+					RuleId: proto.String("string.ulid"),
+				},
+			),
+		},
+		"ulid/invalid/invalid_characters/o": {
+			Message: &cases.StringULID{Val: "01ARZ3NDEKTSV4RRFFQ69G5FAO"},
+			Expected: results.Violations(
+				&validate.Violation{
+					Field:  results.FieldPath("val"),
+					Rule:   results.FieldPath("string.ulid"),
+					RuleId: proto.String("string.ulid"),
+				},
+			),
+		},
+		"ulid/invalid/invalid_characters/u": {
+			Message: &cases.StringULID{Val: "01ARZ3NDEKTSV4RRFFQ69G5FAU"},
+			Expected: results.Violations(
+				&validate.Violation{
+					Field:  results.FieldPath("val"),
+					Rule:   results.FieldPath("string.ulid"),
+					RuleId: proto.String("string.ulid"),
+				},
+			),
+		},
+		"ulid/invalid/exceeds_maximum": {
+			Message: &cases.StringULID{Val: "80000000000000000000000000"},
+			Expected: results.Violations(
+				&validate.Violation{
+					Field:  results.FieldPath("val"),
+					Rule:   results.FieldPath("string.ulid"),
+					RuleId: proto.String("string.ulid"),
+				},
+			),
+		},
+		"ulid/invalid/exceeds_maximum/nine": {
+			Message: &cases.StringULID{Val: "90000000000000000000000000"},
```

Comment:
One exceeding the max should be fine (so keep the 8000... one), but let's replace this one with the max (7ZZZ...).